kv/kvserver: TestReplicaRemovalDuringCPut failed

kv/kvserver.TestReplicaRemovalDuringCPut [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3196521&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3196521&tab=artifacts#/) on master @ [62b6a79c8b2c19b284dbc47b1d4d387d3c4e6d27](https://github.com/cockroachdb/cockroach/commits/62b6a79c8b2c19b284dbc47b1d4d387d3c4e6d27):


```
=== RUN   TestReplicaRemovalDuringCPut
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaRemovalDuringCPut444444167
    test_log_scope.go:74: use -show-logs to present logs inline
    panic.go:965: -- test log scope end --

ERROR: a panic has occurred!
Details cannot be printed yet because we are still unwinding.
Hopefully the test harness prints the panic below, otherwise check the test logs.

test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicaRemovalDuringCPut444444167
    panic.go:965: runtime error: invalid memory address or nil pointer dereference
        goroutine 689641 [running]:
        runtime/debug.Stack(0xc00271f090, 0x3f5ac20, 0x76bf8c0)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:110 +0x27e
        panic(0x3f5ac20, 0x76bf8c0)
        	/usr/local/go/src/runtime/panic.go:965 +0x1b9
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc00e624300, 0x593ea08, 0xc00019e010)
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:466 +0x2d3
        panic(0x3f5ac20, 0x76bf8c0)
        	/usr/local/go/src/runtime/panic.go:965 +0x1b9
        github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).MoveRangeLeaseNonCooperatively.func1(0x3256a9d8, 0xed8878e05)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:982 +0x650
        github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration(0xa7a358200, 0xc00271fa68, 0xc004b29000, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:197 +0xeb
        github.com/cockroachdb/cockroach/pkg/testutils/testcluster.(*TestCluster).MoveRangeLeaseNonCooperatively(0xc011836900, 0x593ea08, 0xc00019e010, 0x28, 0xc012f880d0, 0x1, 0x8, 0xc012f880d8, 0x2, 0x8, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/testcluster/testcluster.go:937 +0x1e5
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.setupReplicaRemovalTest.func2(0x593ea08, 0xc00019e010, 0x5a68688, 0xc008c77290, 0x3, 0x3, 0xc00271fe87)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_relocate_range_test.go:524 +0x43f
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicaRemovalDuringCPut(0xc01406f200)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_relocate_range_test.go:447 +0x462
        testing.tRunner(0xc01406f200, 0x4bc4a30)
        	/usr/local/go/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1238 +0x2b3
--- FAIL: TestReplicaRemovalDuringCPut (0.85s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestReplicaRemovalDuringCPut PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
/cc @cockroachdb/kv erikgrinaker
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicaRemovalDuringCPut.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/testutils/testcluster/testcluster.go
+++ b/pkg/testutils/testcluster/testcluster.go
@@ -969,7 +969,7 @@ func (tc *TestCluster) MoveRangeLeaseNonCooperatively(
 		ls, err := r.TestingAcquireLease(ctx)
 		if err != nil {
 			log.Infof(ctx, "TestingAcquireLease failed: %s", err)
-			if lErr := (*roachpb.NotLeaseHolderError)(nil); errors.As(err, &lErr) {
+			if lErr := (*roachpb.NotLeaseHolderError)(nil); errors.As(err, &lErr) && lErr.Lease != nil {
 				newLease = lErr.Lease
 			} else {
 				return err

